File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_private_clause.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>
#include <openacc.h>

#ifndef T1
//T1:loop construct private clause,V:2.7-3.3
int test1(){
    int err = 0;
    srand(time(0));

    const int n = 100;
    int a[n], b[n], c[n], tmp;

    // Initialize arrays
    for (int i = 0; i < n; ++i) {
        b[i] = rand() % 100;
        c[i] = rand() % 100;
    }

    #pragma acc data copyin(b, c) copyout(a)
    {
        #pragma acc parallel loop gang worker private(tmp)
        for (int k = 0; k < n; ++k) {
            tmp = b[k] + c[k];
            #pragma acc loop vector
            for (int i = 1; i < n; ++i) {
                a[i] = a[i] + tmp;
            }
        }
    }

    // Check the results
    for (int i = 0; i < n; ++i) {
        if (a[i] != b[i] + c[i] + tmp) {
            err = 1;
            break;
        }
    }

    return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < 10; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: Deepseek-Coder-33b-Instruct/stage2/loop_construct_private_clause.f90
!T1:loop construct private clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, PARAMETER :: N = 100
        REAL(KIND=8), DIMENSION(N) :: a, b, c
        INTEGER :: i, j, k, div
        REAL(KIND=8) :: tmp

        ! Initialize arrays
        a = 0.0
        b = 1.0
        c = 2.0
        div = 3

        !$acc data copyin(a, b, c, div) copyout(a)
        !$acc parallel loop gang
        do k = 1, N
          !$acc loop worker private(tmp)
          do j = 1, N
            tmp = b(j) + c(j)
            !$acc loop vector
            do i = 1, N
              a(i) = a(i) + tmp/div
            enddo
          enddo
        enddo
        !$acc end data

        ! Check the results
        do i = 1, N
          if (a(i) .ne. b(i) + c(i) / div) then
            errors = errors + 1
          endif
        enddo

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM